File: java-selenium/src/test/java/com/orangehrm/tests/DashboardTest.java
package com.orangehrm.tests;

import com.orangehrm.base.BaseTest;
import com.orangehrm.pages.DashboardPage;
import com.orangehrm.pages.LoginPage;
import com.orangehrm.pages.AdminPage;
import com.orangehrm.pages.PIMPage;
import com.orangehrm.utils.ConfigReader;
import org.testng.Assert;
import org.testng.annotations.Test;

public class DashboardTest extends BaseTest {

    @Test(priority = 1, description = "Verify dashboard navigation to Admin page")
    public void testNavigateToAdminPage() {
        LoginPage loginPage = new LoginPage(driver);
        DashboardPage dashboardPage = loginPage.login(
            ConfigReader.getUsername(),
            ConfigReader.getPassword()
        );
        
        // Navigate to Admin page
        dashboardPage.clickAdminMenu();
        AdminPage adminPage = new AdminPage(driver);
        
        // Verify Admin page is displayed
        Assert.assertTrue(adminPage.isAdminPageDisplayed(), "Admin page should be displayed");
        Assert.assertEquals(adminPage.getAdminPageTitle(), "Admin", "Admin page title should match");
    }

    @Test(priority = 2, description = "Verify dashboard navigation to PIM page")
    public void testNavigateToPIMPage() {
        LoginPage loginPage = new LoginPage(driver);
        DashboardPage dashboardPage = loginPage.login(
            ConfigReader.getUsername(),
            ConfigReader.getPassword()
        );
        
        // Navigate to PIM page
        dashboardPage.clickPIMMenu();
        PIMPage pimPage = new PIMPage(driver);
        
        // Verify PIM page is displayed
        Assert.assertTrue(pimPage.isPIMPageDisplayed(), "PIM page should be displayed");
        Assert.assertEquals(pimPage.getPIMPageTitle(), "PIM", "PIM page title should match");
    }

    @Test(priority = 3, description = "Verify logout functionality")
    public void testLogout() {
        LoginPage loginPage = new LoginPage(driver);
        DashboardPage dashboardPage = loginPage.login(
            ConfigReader.getUsername(),
            ConfigReader.getPassword()
        );
        
        // Verify dashboard is displayed
        Assert.assertTrue(dashboardPage.isDashboardDisplayed(), "Dashboard should be displayed");
        
        // Perform logout
        LoginPage logoutPage = dashboardPage.logout();
        
        // Verify login page is displayed after logout
        Assert.assertTrue(logoutPage.isLoginPageDisplayed(), "Login page should be displayed after logout");
    }

    @Test(priority = 4, description = "Verify all main menu items are present")
    public void testMainMenuItems() {
        LoginPage loginPage = new LoginPage(driver);
        DashboardPage dashboardPage = loginPage.login(
            ConfigReader.getUsername(),
            ConfigReader.getPassword()
        );
        
        // Verify dashboard is displayed
        Assert.assertTrue(dashboardPage.isDashboardDisplayed(), "Dashboard should be displayed");
        
        // Verify main menu items can be clicked (basic presence check)
        // Note: In a real scenario, you might want to verify menu items are visible
        dashboardPage.clickAdminMenu();
        AdminPage adminPage = new AdminPage(driver);
        Assert.assertTrue(adminPage.isAdminPageDisplayed(), "Admin menu should navigate correctly");
    }
}

File: playwright-typescript/tests/dashboard.test.ts
import { test, expect } from '@playwright/test';
import { LoginPage } from '../src/pages/login-page';
import { DashboardPage } from '../src/pages/dashboard-page';
import { AdminPage } from '../src/pages/admin-page';
import { PIMPage } from '../src/pages/pim-page';
import ConfigReader from '../src/utils/config';

test.describe('Dashboard Tests', () => {
  test.beforeEach(async ({ page }) => {
    await page.goto(ConfigReader.getBaseUrl());
  });

  test('Verify dashboard navigation to Admin page', async ({ page }) => {
    const loginPage = new LoginPage(page);
    const dashboardPage: DashboardPage = await loginPage.login(
      ConfigReader.getUsername(),
      ConfigReader.getPassword()
    );
    
    // Navigate to Admin page
    await dashboardPage.clickAdminMenu();
    const adminPage = new AdminPage(page);
    
    // Verify Admin page is displayed
    await expect(adminPage.isAdminPageDisplayed()).resolves.toBe(true);
    await expect(adminPage.getAdminPageTitle()).resolves.toBe('Admin');
  });

  test('Verify dashboard navigation to PIM page', async ({ page }) => {
    const loginPage = new LoginPage(page);
    const dashboardPage: DashboardPage = await loginPage.login(
      ConfigReader.getUsername(),
      ConfigReader.getPassword()
    );
    
    // Navigate to PIM page
    await dashboardPage.clickPIMMenu();
    const pimPage = new PIMPage(page);
    
    // Verify PIM page is displayed
    await expect(pimPage.isPIMPageDisplayed()).resolves.toBe(true);
    await expect(pimPage.getPIMPageTitle()).resolves.toBe('PIM');
  });

  test('Verify logout functionality', async ({ page }) => {
    const loginPage = new LoginPage(page);
    const dashboardPage: DashboardPage = await loginPage.login(
      ConfigReader.getUsername(),
      ConfigReader.getPassword()
    );
    
    // Verify dashboard is displayed
    await expect(dashboardPage.isDashboardDisplayed()).resolves.toBe(true);
    
    // Perform logout
    const logoutPage = await dashboardPage.logout();
    
    // Verify login page is displayed after logout
    await expect(logoutPage.isLoginPageDisplayed()).resolves.toBe(true);
  });

  test('Verify all main menu items are present', async ({ page }) => {
    const loginPage = new LoginPage(page);
    const dashboardPage: DashboardPage = await loginPage.login(
      ConfigReader.getUsername(),
      ConfigReader.getPassword()
    );
    
    // Verify dashboard is displayed
    await expect(dashboardPage.isDashboardDisplayed()).resolves.toBe(true);
    
    // Verify main menu items can be clicked (basic presence check)
    // Note: In a real scenario, you might want to verify menu items are visible
    await dashboardPage.clickAdminMenu();
    const adminPage = new AdminPage(page);
    await expect(adminPage.isAdminPageDisplayed()).resolves.toBe(true);
  });
});

